src/main/java/org/breedinginsight/model/delta/DeltaEntityFactory.java

```diff
@@ -0,0 +1,93 @@
+package org.breedinginsight.model.delta;
+
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Prototype;
+import org.breedinginsight.model.delta.DeltaGermplasm;
+import org.breedinginsight.model.delta.DeltaObservation;
+import org.breedinginsight.model.delta.DeltaObservationUnit;
+import org.breedinginsight.model.delta.DeltaObservationVariable;
+import org.breedinginsight.model.delta.Environment;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.brapi.v2.model.core.BrAPILocation;
+import org.brapi.v2.model.core.BrAPIStudy;
+import org.brapi.v2.model.core.BrAPITrial;
+import org.brapi.v2.model.germ.BrAPIGermplasm;
+import org.brapi.v2.model.pheno.BrAPIObservation;
+import org.brapi.v2.model.pheno.BrAPIObservationUnit;
+import org.brapi.v2.model.pheno.BrAPIObservationVariable;
+
+@Factory
+public class DeltaEntityFactory {
+
+    public static Experiment makeExperiment(BrAPITrial brAPIObject) {
+        return new Experiment(brAPIObject);
+    }
+
+    public static Environment makeEnvironment(BrAPIStudy brAPIObject) {
+        return new Environment(brAPIObject);
+    }
+
+    public static DeltaGermplasm makeDeltaGermplasm(BrAPIGermplasm brAPIObject) {
+        return new DeltaGermplasm(brAPIObject);
+    }
+
+    public static DeltaLocation makeDeltaLocation(BrAPILocation brAPIObject) {
+        return new DeltaLocation(brAPIObject);
+    }
+
+    public static DeltaObservation makeDeltaObservation(BrAPIObservation brAPIObject) {
+        return new DeltaObservation(brAPIObject);
+    }
+
+    public static DeltaObservationUnit makeDeltaObservationUnit(BrAPIObservationUnit brAPIObject) {
+        return new DeltaObservationUnit(brAPIObject);
```

Comment:
Include the ObservationUnitService as an arg here.

Reply:
👍 Done.